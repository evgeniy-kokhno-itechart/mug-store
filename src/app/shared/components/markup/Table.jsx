import React from 'react';
import { PropTypes } from 'prop-types';
import TableHeader from './TableHeader';
import TableBody from './TableBody';

const Table = (props) => {
  const { columns, items, customClasses } = props;
  return (
    <div className="table-responsive-md">
      <table className={`table ${customClasses}`}>
        <TableHeader columns={columns} />
        <TableBody items={items} columns={columns} />
      </table>
    </div>
  );
};

Table.propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  customClasses: PropTypes.string,
};

Table.defaultProps = {
  customClasses: '',
};

export default Table;
