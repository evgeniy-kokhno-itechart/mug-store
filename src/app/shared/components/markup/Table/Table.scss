.info-table {
    border-collapse: collapse;
}