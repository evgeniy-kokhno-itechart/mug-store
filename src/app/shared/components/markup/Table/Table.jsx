import React from 'react';
import { PropTypes } from 'prop-types';
import TableHeader from '../TableHeader/TableHeader';
import TableBody from '../TableBody/TableBody';
import './Table.scss';

const Table = ({
  columns, items, children, customClasses,
}) => (
  <table className={`info-table ${customClasses}`}>
    <TableHeader columns={columns} />
    <TableBody items={items} columns={columns} />
    {children}
  </table>
);

Table.propTypes = {
  columns: PropTypes.arrayOf(PropTypes.object).isRequired,
  items: PropTypes.arrayOf(PropTypes.object).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  customClasses: PropTypes.string,
};

Table.defaultProps = {
  customClasses: '',
  children: null,
};

export default Table;
