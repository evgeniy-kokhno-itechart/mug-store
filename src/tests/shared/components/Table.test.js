/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Table from '../../../app/shared/components/markup/Table/Table';

configure({ adapter: new Adapter() });

describe('<Table />', () => {
  let TableWrapper;

  beforeEach(() => {
    TableWrapper = shallow(
      <Table
        items={[
          { id: 1, name: 'one', description: 'first' },
          { id: 2, name: 'two', description: 'second' },
          { id: 3, name: 'three', description: 'third' },
        ]}
        columns={[
          { path: 'id', label: 'ID' },
          { path: 'name', label: 'Name' },
          { path: 'description', label: 'Description', content: item => <span>{`Test Content ${item.description}`}</span> },
        ]}
      />,
    );
  });

  it('renders properly with default props', () => {
    expect(TableWrapper).toMatchSnapshot();
  });

  it('renders without header when headerExcluded is true', () => {
    TableWrapper.setProps({ headerExcluded: true });
    expect(TableWrapper).toMatchSnapshot();
  });

  it('renders within header when headerExcluded is false', () => {
    TableWrapper.setProps({ headerExcluded: false });
    expect(TableWrapper).toMatchSnapshot();
  });
});
