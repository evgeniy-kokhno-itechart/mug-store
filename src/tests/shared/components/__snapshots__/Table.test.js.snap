// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders properly with default props 1`] = `
<table
  className="info-table "
>
  <TableHeader
    columns={
      Array [
        Object {
          "label": "ID",
          "path": "id",
        },
        Object {
          "label": "Name",
          "path": "name",
        },
        Object {
          "content": [Function],
          "label": "Description",
          "path": "description",
        },
      ]
    }
  />
  <TableBody
    columns={
      Array [
        Object {
          "label": "ID",
          "path": "id",
        },
        Object {
          "label": "Name",
          "path": "name",
        },
        Object {
          "content": [Function],
          "label": "Description",
          "path": "description",
        },
      ]
    }
    items={
      Array [
        Object {
          "description": "first",
          "id": 1,
          "name": "one",
        },
        Object {
          "description": "second",
          "id": 2,
          "name": "two",
        },
        Object {
          "description": "third",
          "id": 3,
          "name": "three",
        },
      ]
    }
  />
</table>
`;

exports[`<Table /> renders within header when headerExcluded is false 1`] = `
<table
  className="info-table "
>
  <TableHeader
    columns={
      Array [
        Object {
          "label": "ID",
          "path": "id",
        },
        Object {
          "label": "Name",
          "path": "name",
        },
        Object {
          "content": [Function],
          "label": "Description",
          "path": "description",
        },
      ]
    }
  />
  <TableBody
    columns={
      Array [
        Object {
          "label": "ID",
          "path": "id",
        },
        Object {
          "label": "Name",
          "path": "name",
        },
        Object {
          "content": [Function],
          "label": "Description",
          "path": "description",
        },
      ]
    }
    items={
      Array [
        Object {
          "description": "first",
          "id": 1,
          "name": "one",
        },
        Object {
          "description": "second",
          "id": 2,
          "name": "two",
        },
        Object {
          "description": "third",
          "id": 3,
          "name": "three",
        },
      ]
    }
  />
</table>
`;

exports[`<Table /> renders without header when headerExcluded is true 1`] = `
<table
  className="info-table "
>
  <TableHeader
    columns={
      Array [
        Object {
          "label": "ID",
          "path": "id",
        },
        Object {
          "label": "Name",
          "path": "name",
        },
        Object {
          "content": [Function],
          "label": "Description",
          "path": "description",
        },
      ]
    }
  />
  <TableBody
    columns={
      Array [
        Object {
          "label": "ID",
          "path": "id",
        },
        Object {
          "label": "Name",
          "path": "name",
        },
        Object {
          "content": [Function],
          "label": "Description",
          "path": "description",
        },
      ]
    }
    items={
      Array [
        Object {
          "description": "first",
          "id": 1,
          "name": "one",
        },
        Object {
          "description": "second",
          "id": 2,
          "name": "two",
        },
        Object {
          "description": "third",
          "id": 3,
          "name": "three",
        },
      ]
    }
  />
</table>
`;
